api/configuration/v1alpha1/konnect_controlplaneref_types.go

```diff
@@ -8,18 +8,26 @@ const (
 	// It is used to reference a Konnect Control Plane entity inside the cluster
 	// using a namespaced reference.
 	ControlPlaneRefKonnectNamespacedRef = "konnectNamespacedRef"
+	// ControlPlaneRefKIC is the type for KIC ControlPlaneRef.
+	// It is used to reference a KIC as Control Plane.
+	ControlPlaneRefKIC = "kic"
 )
 
 // ControlPlaneRef is the schema for the ControlPlaneRef type.
 // It is used to reference a Control Plane entity.
 // +kubebuilder:validation:XValidation:rule="self.type == 'konnectNamespacedRef' ? has(self.konnectNamespacedRef) : true", message="when type is konnectNamespacedRef, konnectNamespacedRef must be set"
+// +kubebuilder:validation:XValidation:rule="self.type == 'konnectNamespacedRef' ? !has(self.konnectID) : true", message="when type is konnectNamespacedRef, konnectID must not be set"
 // +kubebuilder:validation:XValidation:rule="self.type == 'konnectID' ? has(self.konnectID) : true", message="when type is konnectID, konnectID must be set"
+// +kubebuilder:validation:XValidation:rule="self.type == 'konnectID' ? !has(self.konnectNamespacedRef) : true", message="when type is konnectID, konnectNamespacedRef must be set"
```

Comment:
```suggestion
// +kubebuilder:validation:XValidation:rule="self.type == 'konnectID' ? !has(self.konnectNamespacedRef) : true", message="when type is konnectID, konnectNamespacedRef must not be set"
```
Please also check the expected messages in test cases.

Reply:
Thanks for catching this 🙇 

e3cc89393336c8b8cb86f32ad3e65a74760c5676 should handle this.